sdf/final-project/technical.md

```diff
@@ -6,102 +6,138 @@ This is intended to be filled out by staff members of DPI Tech Apprenticeship pr
+- [ ] **Validations**: Are validations implemented to ensure data integrity and consistency?
+- [ ] **Query Optimization**: Does the application use scopes to perform optimized database queries?
 - [ ] **Database Management**: Are additional features such as file upload (CSV) or custom rake tasks for database management included?
+- [ ] **Testing**: Does the project include a comprehensive test suite (e.g., RSpec, Minitest) that covers models, controllers, and other critical components?
+
+### Notes:
 
-### Score (/5):
+## Security and Authorization (max: 5 points)
+- [ ] **Credentials**: Are API keys and sensitive information securely stored (using .env or Rails credentials)?
+- [ ] **HTTPS**: Is HTTPS enforced? (config.force_ssl = true)?
+- [ ] **Sensitive attributes**: Are sensitive attributes assigned in the model or controller when necessary (e.g. current_user), and not through hidden fields?
+- [ ] **Strong Params**: Are strong parameters used to prevent form vulnerabilities?
+- [ ] **Authorization**: Is an authorization framework (such as Pundit) employed to manage user permissions and ensure secure access control throughout the application?
 
 ### Notes:
 
-## Features (max 5pts)
+## Ambitious Features (each: 2 points - max: 10 points)
+- [ ] **Receiving Email**: Implementation of ActionMailbox to receive emails within the application.
+- [ ] **Web Scraping Capabilities**: Incorporation of web scraping functionality to extract data from external websites.
+- [ ] **Background Processing**: Are background jobs (eg ActiveJob) implemented for time-consuming processes, improving app performance and user experience?
+- [ ] **Mapping and Geolocation**: Use of mapping or geocoding libraries (e.g., Mapbox, Geocoder) to add location-based features to the application.
 - [ ] **Cloud Storage Integration**: Integration with cloud storage services (e.g., AWS S3) for handling file uploads and storage.
+- [ ] **Chat GPT or AI Integration**: Implementation of Chat GPT or other AI services to provide intelligent responses or features. 
 - [ ] **Payment Processing**: Implementation of a payment gateway (e.g., Stripe) for secure online transactions.
-- [ ] **Custom Email Functionality**: Use of mailers (either through Devise or custom implementations) to send emails for account confirmation, password recovery, notifications, etc.
-- [ ] **Advanced Search and Filtering**: Incorporation of advanced search and filtering capabilities (e.g., using Ransack) to improve data retrieval and user experience.
-- [ ] **Mapping and Geolocation**: Use of mapping or geocoding libraries (e.g., Mapbox, Geocoder) to add location-based features to the application.
 - [ ] **Hotwire**: Implementation of Hotwire, Turbo, or Stimulus to create a dynamic, highly interactive user interface without relying heavily on traditional full-page reloads.
```

Comment:
Thoughts on putting stimulus into enrichment bucket since it's now included in the javascript module? Hotwire is the brand name around turbo, stimulus, and strada. If we want to keep Turbo here, I think we should be explicit that we mean Turbo Drive / Turbo Frames / Turbo Streams (not turbo drive which is baked into Rails by default)

Reply:
added and moved to enrichment